src/NuGet.Core/NuGet.Test.Utility/CommandRunner.cs

```diff
@@ -58,36 +59,19 @@ public class CommandRunner
             {
                 p = new Process();
 
-                p.OutputDataReceived += (o, e) =>
-                {
-                    if (e.Data != null)
-                    {
-                        output.AppendLine(e.Data);
-                    }
-                };
-
-                p.ErrorDataReceived += (o, e) =>
-                {
-                    if (e.Data != null)
-                    {
-                        errors.AppendLine(e.Data);
-                    }
-                };
-
                 p.StartInfo = psi;
                 p.Start();
 
-                p.BeginErrorReadLine();
-                p.BeginOutputReadLine();
+                var outputTask = ConsumeStreamReaderAsync(p.StandardOutput, output);
+                var errorTask = ConsumeStreamReaderAsync(p.StandardError, errors);
```

Comment:
Does this need to run under Task.Run()? Since down farther Task.WaitAll is not async I don't see how this will actually allow multi threading and switch between the threads, it just works on its own?